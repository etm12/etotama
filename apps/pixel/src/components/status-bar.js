import * as React from 'karet';
import * as U from 'karet.util';

const classPrefix = 'c-status-bar';

export const StatusLabel = ({ className, children }) => {
  const cname = U.string`${classPrefix}__label`;
  return (
    <div
      className={U.cns(
        cname,
        className,
      )}>
      {children}
    </div>
  );
};

export const StatusIndicator = ({ className, label, active }) => {
  const cname = U.string`${classPrefix}__indicator`;

  return (
    <div className={U.cns(
      cname,
      U.when(active, U.string`${cname}--active`),
      U.string`${classPrefix}-item`,
      className,
    )}>
      {label}
    </div>
  );
}

export const StatusBar = ({ className, children }) => {
  return (
    <div
      className={U.cns(
        classPrefix,
        className,
      )}
    >
      Status
    </div>
  );
};

export default StatusBar;
