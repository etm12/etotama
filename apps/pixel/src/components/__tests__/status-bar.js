import * as React from 'karet';
import { mount } from 'enzyme';
import * as SB from '../status-bar';

describe('StatusBar', () => {;
  describe.each(Object.entries(SB).filter(([k]) => k !== 'default'))('StatusBar.%s', (k, Component) => {
    test('mounts without crashing', () => {
      mount(<Component />);
    });

    test('has not inadvertently changed', () => {
      Component.displayName = k;

      expect(mount(<Component />).children().first()).toMatchSnapshot();
    });
  });
});
