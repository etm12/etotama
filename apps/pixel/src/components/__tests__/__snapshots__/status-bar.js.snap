// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar StatusBar.StatusBar has not inadvertently changed 1`] = `
<StatusBar>
  <div
    className="c-status-bar"
  >
    Status
  </div>
</StatusBar>
`;

exports[`StatusBar StatusBar.StatusGroup has not inadvertently changed 1`] = `
<StatusGroup>
  <div
    className="c-status-bar__group"
  />
</StatusGroup>
`;

exports[`StatusBar StatusBar.StatusIndicator has not inadvertently changed 1`] = `
<StatusIndicator>
  <div
    className="c-status-bar__indicator c-status-bar-item"
  />
</StatusIndicator>
`;

exports[`StatusBar StatusBar.StatusLabel has not inadvertently changed 1`] = `
<StatusLabel>
  <div
    className="c-status-bar__label"
  />
</StatusLabel>
`;
